x-pack/solutions/security/plugins/security_solution/public/cases/components/ai_for_soc/table.tsx

```diff
@@ -0,0 +1,126 @@
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { memo, useMemo } from 'react';
+import type { DataView } from '@kbn/data-views-plugin/common';
+import { AlertsTable } from '@kbn/response-ops-alerts-table';
+import type { PackageListItem } from '@kbn/fleet-plugin/common';
+import type { QueryDslQueryContainer } from '@elastic/elasticsearch/lib/api/types';
+import type { Alert } from '@kbn/alerting-types';
+import type { EuiDataGridColumn } from '@elastic/eui';
+import type { AdditionalTableContext } from '../../../detections/components/alert_summary/table/table';
+import {
+  ACTION_COLUMN_WIDTH,
+  ALERT_TABLE_CONSUMERS,
+  columns,
+  GRID_STYLE,
+  ROW_HEIGHTS_OPTIONS,
+  RULE_TYPE_IDS,
+  TOOLBAR_VISIBILITY,
+} from '../../../detections/components/alert_summary/table/table';
+import { ActionsCell } from '../../../detections/components/alert_summary/table/actions_cell';
+import { getDataViewStateFromIndexFields } from '../../../common/containers/source/use_data_view';
+import { useKibana } from '../../../common/lib/kibana';
+import { CellValue } from '../../../detections/components/alert_summary/table/render_cell';
+import type { RuleResponse } from '../../../../common/api/detection_engine';
+
+export interface TableProps {
+  /**
+   * DataView created for the alert summary page
+   */
+  dataView: DataView;
+  /**
+   * Id to pass down to the ResponseOps alerts table
+   */
+  id: string;
+  /**
+   * Callback fired when the alerts have been first loaded
+   */
+  onLoaded?: (alerts: Alert[], columns: EuiDataGridColumn[]) => void;
+  /**
+   * List of installed AI for SOC integrations
+   */
+  packages: PackageListItem[];
+  /**
+   * Query that contains the id of the alerts to display in the table
+   */
+  query: Pick<QueryDslQueryContainer, 'bool' | 'ids'>;
+  /**
+   * Result from the useQuery to fetch all rules
+   */
+  ruleResponse: {
+    /**
+     * Result from fetching all rules
+     */
+    rules: RuleResponse[];
+    /**
+     * True while rules are being fetched
+     */
+    isLoading: boolean;
+  };
+}
+
+/**
+ * Component used in the Cases page under Alerts tab, only in the AI4DSOC tier.
+ * It leverages a lot of configurations and constants from the Alert summary page alerts table, and renders the ResponseOps AlertsTable.
+ */
+export const Table = memo(
+  ({ dataView, id, onLoaded, packages, query, ruleResponse }: TableProps) => {
+    const {
+      services: { application, data, fieldFormats, http, licensing, notifications, settings },
+    } = useKibana();
+    const services = useMemo(
+      () => ({
+        data,
+        http,
+        notifications,
+        fieldFormats,
+        application,
+        licensing,
+        settings,
+      }),
+      [application, data, fieldFormats, http, licensing, notifications, settings]
+    );
+
+    const dataViewSpec = useMemo(() => dataView.toSpec(), [dataView]);
+
+    const { browserFields } = useMemo(
```

Comment:
says `browserFields` is deprecated
![image](https://github.com/user-attachments/assets/51368ccc-237f-4ee7-9f65-366287aa56e4)

Reply:
It does but the ResponseOps alerts table needs it :man-shrugging:
It's been deprecated for 2 years and nothing to replace it by yet...